lib/facility.rb

```diff
@@ -1,14 +1,79 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees, 
+              :daily_hours, 
+              :holidays
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
+    @daily_hours = facility_details[:daily_hours]
+    @holidays = facility_details[:holidays]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if can_perform?('Vehicle Registration')
+      if vehicle.antique? 
+        @collected_fees += 25 
+        vehicle.change_plate_type(:antique)
+      elsif vehicle.electric_vehicle? 
+        @collected_fees += 200 
+        vehicle.change_plate_type(:ev)
+      else
+        @collected_fees += 100 
+        vehicle.change_plate_type(:regular)
+      end
+      vehicle.change_registration_date
+      @registered_vehicles << vehicle 
+    end
+  end
```

Comment:
This nested flow control is where things can get confusing and slowed down. What if instead of that outer if statement, you used a guard clause? Something like `return nil if !can_perform("Vehicle Registration")`. That would be a single line guard clause to first just check that the facility can do the action. And if it can't, nil will be returned right away and it won't go down the rest of the method. Then you're next if statement can be it's own block and not nested.